src/app/InteractionModelEngine.cpp

```diff
@@ -530,10 +532,12 @@ CHIP_ERROR InteractionModelEngine::OnUnsolicitedReportData(Messaging::ExchangeCo
             continue;
         }
 
-        return readClient->OnUnsolicitedReportData(apExchangeContext, std::move(aPayload));
+        VerifyOrReturnError(readClient->OnUnsolicitedReportData(apExchangeContext, std::move(aPayload)) == CHIP_NO_ERROR,
+                            Status::InvalidAction);
+        return Status::Success;
     }
 
-    return CHIP_NO_ERROR;
+    return Status::InvalidAction;
```

Comment:
Why is this InvalidAction?  Per spec this should be InvalidSubscription, no?  I guess this is tracked in https://github.com/project-chip/connectedhomeip/issues/19130 but if we're here anyway we might as well fix it.  This was correct in #19356 and looks like it was changed when changing the "parse failures" to InvalidAction... but this is not a "parse failure".

Reply:
fixed, thanks